File: java-bridge-matlab/src/main/java/idesyde/matlab/MatlabIdentificationModule.java
package idesyde.matlab;

import idesyde.blueprints.IdentificationModuleBlueprint;
import idesyde.core.DecisionModel;
import idesyde.core.IdentificationModule;
import idesyde.core.IdentificationRule;
import idesyde.core.ReverseIdentificationRule;
import idesyde.core.headers.DecisionModelHeader;

import java.util.Optional;
import java.util.Set;

public class MatlabIdentificationModule implements IdentificationModuleBlueprint {
    @Override
    public String uniqueIdentifier() {
        return "MatlabIdentificationModule";
    }

    @Override
    public Optional<DecisionModel> decisionHeaderToModel(DecisionModelHeader header) {
        return Optional.empty();
    }

    @Override
    public Set<ReverseIdentificationRule> reverseIdentificationRules() {
        return Set.of();
    }

    @Override
    public Set<IdentificationRule> identificationRules() {
        return Set.of(
                new CommunicatingAndTriggeredReactiveWorkloadIRule()
        );
    }
}

File: kotlin-bridge-matlab/src/main/kotlin/idesyde/matlab/MatlabIdentificationModule.kt
package idesyde.matlab

import idesyde.core.*

object MatlabIdentificationModule : IdentificationModule {
    override fun uniqueIdentifier(): String = "MatlabIdentificationModule"

    override fun decisionHeaderToModel(m: DecisionModelHeader): DecisionModel? {
        TODO("Not yet implemented")
    }

    override fun designHeaderToModel(m: DesignModelHeader): DesignModel? {
        TODO("Not yet implemented")
    }

    override fun reverseIdentificationRules(): Set<(Set<DesignModel>, Set<DecisionModel>) -> Set<DesignModel>> {
        TODO("Not yet implemented")
    }

    override fun identificationRules(): Set<IdentificationRule<DecisionModel>> = setOf(
        CommunicatingAndTriggeredReactiveWorkloadIRule()
    )

    @JvmStatic
    fun main(args: Array<String>): Unit {
        standaloneIdentificationModule(args)
    }
}